examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,489 @@
+)
+
+"""
+# 7. Finetune the model
+
+Finetuning is an art when it comes to Machine Learning, and there are many ways to adjust
+the model in efforts to improve it. Finetuning is beyond the scope of this notebook, but
+check out this
+[article](https://www.pyimagesearch.com/2019/06/03/fine-tuning-with-keras-and-deep-learnin
+g/) for more information.
+
+For our purposes, we'll use Keras callbacks to further finetune our model. The checkpoint
+callback saves the best weights of the model, so next time we want to use the model, we
+do not have to spend time training it. The early stopping callback stops the training
+process when the model starts becoming stagnant, or even worse, when the model starts
+overfitting. Since we set `restore_best_weights` to `True`, the returned model at the end
+of the training process will be the model with the best weights (i.e. low loss and high
+accuracy).
+"""
+
+checkpoint_cb = tf.keras.callbacks.ModelCheckpoint("xray_model.h5", save_best_only=True)
+
+early_stopping_cb = tf.keras.callbacks.EarlyStopping(
+    patience=10, restore_best_weights=True
+)
+
+"""
+We also want to finetune our learning rate. Too high of a learning rate will cause the
+model to diverge. Too small of a learning rate will cause the model to be too slow. We
+implement the exponential learning rate scheduling method below.
+"""
+
+
+def exponential_decay(lr0, s):
+    def exponential_decay_fn(epoch):
+        return lr0 * 0.1 ** (epoch / s)
+
+    return exponential_decay_fn
+
+
+exponential_decay_fn = exponential_decay(0.01, 20)
+
+lr_scheduler = tf.keras.callbacks.LearningRateScheduler(exponential_decay_fn)
+
+history = model.fit(
+    train_ds,
+    steps_per_epoch=TRAIN_IMG_COUNT // BATCH_SIZE,
+    epochs=100,
+    validation_data=val_ds,
+    validation_steps=VAL_IMG_COUNT // BATCH_SIZE,
+    class_weight=class_weight,
+    callbacks=[checkpoint_cb, early_stopping_cb, lr_scheduler],
+)
+
+"""
+# 8. Visualizing model performance
+
+Let's plot the model accuracy and loss for the training and the validating set. These
+plots show the accuracy and loss values for the second round of training. Since we
+initially trained the model with 30 epochs, these would be epochs 31-45. Note that no
+random seed is specified for this notebook. For your notebook, there might be slight
+variance.
+"""
+
+fig, ax = plt.subplots(1, 4, figsize=(20, 3))
+ax = ax.ravel()
+
+for i, met in enumerate(["precision", "recall", "accuracy", "loss"]):
+    ax[i].plot(history.history[met])
+    ax[i].plot(history.history["val_" + met])
+    ax[i].set_title("Model {}".format(met))
+    ax[i].set_xlabel("epochs")
+    ax[i].set_ylabel(met)
+    ax[i].legend(["train", "val"])
+
+"""
+We see that the accuracy for our model is around 98%. Finetune the model further to see
+if we can achieve a higher accuracy.
+"""
+
+"""
+# 9. Predict and evaluate results
+
+Let's evaluate the model on our test data!
+"""
+
+loss, acc, prec, rec = model.evaluate(test_ds)
```

Comment:
Since you have many quantities, prefer using `evaluate(test_ds, return_dict=True)`